Sources/Datadog/DatadogInternal/Concurrency/Concurrency.swift

```diff
@@ -0,0 +1,56 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+import Foundation
+
+/// A property wrapper using a fair, POSIX conforming reader-writer lock for atomic
+/// access to the value.
```

Comment:
I'm really missing this important context, WDYT?
```suggestion
/// A property wrapper using a fair, POSIX conforming reader-writer lock for atomic
/// access to the value. It is optimised for concurrent reads and exclusive writes.
```